test/foreman_tasks_test_helper.rb

```diff
@@ -7,13 +7,16 @@
 require_relative './support/history_tasks_builder'
 
 require 'dynflow/testing'
+require 'foreman_tasks/test_helpers'
 
 FactoryBot.definition_file_paths = ["#{ForemanTasks::Engine.root}/test/factories"]
 FactoryBot.find_definitions
 
 ForemanTasks.dynflow.require!
 ForemanTasks.dynflow.config.disable_active_record_actions = true
 
+ForemanTasks::TestHelpers.use_in_memory_sqlite!
```

Comment:
Seems like using the same db for standard test world and the in-thread executor one was causing issues: this should make the things more isolated.